### PR TITLE
Pre production

### PR DESCRIPTION
**Pull request by MaxDaniLOVE**
### Implemented:
- Add feature of sending e-mail;
- Fix minor issues of the previous PR;
- Fix styles;
- [Demo](https://dims-5.netlify.app/)
logins (ADMIN: admin@gmail.com, MENTOR: mentor@gmail.com, USER: mahseeen@gmail.com)
pass: 12345678

### CODE REVIEW

File: src/UI/CurrentUser/CurrentUser.js [hunk 1]
```diff
@@ -1,8 +1,9 @@
 import React from 'react';
 import PropTypes from 'prop-types';
+import getMailLogin from '../../utils/getMailLogin';
 import './currentUser.scss';
 
-const CurrentUser = ({ children }) => <div className='current-user'>{children}</div>;
+const CurrentUser = ({ children }) => <div className='current-user'>{getMailLogin(children)}</div>;
```
Comment (maximsan) [inline src/UI/CurrentUser/CurrentUser.js:6]: it's better to pass here already prepared children and your component would be less coupled with concrete method

File: src/UI/CustomLinks/SocialNetworksLink.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import Link from './Link';
+
+const SocialNetworksLink = ({ children, href }) => {
+  return (
+    <Link href={href} target='_blank' rel='noopener noreferrer'>
```
Comment (maximsan) [inline src/UI/CustomLinks/SocialNetworksLink.js:7]: mover rel and target to default props

File: src/components/InTouchForm/InTouchForm.js [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+    setFormData(updated);
+    setIsFormValid(isValid);
+  };
+
+  const sendMessageToAuthor = async () => {
+    startSending();
+    try {
+      await sendMail(formData);
+    } catch (error) {
+      stopSending();
```
Comment (maximsan) [inline src/components/InTouchForm/InTouchForm.js:56]: where is error handling ? notification ? user should know if something has gone wrong

File: src/components/InTouchForm/InTouchForm.js [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+    const value = formData[id];
+    return (
+      <AvGroup className='form-inputs' key={id}>
+        <Label htmlFor={id}>
+          {label}
+          <AvField name={id} value={value} type={type} id={id} onChange={onChange} validate={pattern} />
+        </Label>
+      </AvGroup>
+    );
+  });
```
Comment (maximsan) [inline src/components/InTouchForm/InTouchForm.js:73]: it can be component

File: src/components/InTouchForm/InTouchForm.js [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+
+  const defaultClassName = 'modal-window intouch-form';
```
Comment (maximsan) [inline src/components/InTouchForm/InTouchForm.js:75]: it can be moved out of component

File: src/components/InTouchForm/InTouchForm.js [hunk 6]
```diff
@@ -0,0 +1,113 @@
...
+
+  const openModal = () => {
+    setIsShowModal(true);
+  };
+  const closeModal = () => {
+    setIsShowModal(false);
+    setTimeout(() => {
+      setFormData(defaultInTouchData);
+    }, 150);
+  };
```
Comment (maximsan) [inline src/components/InTouchForm/InTouchForm.js:33]: it can be wrapped with useCallback

File: src/components/InTouchForm/InTouchForm.js [hunk 7]
```diff
@@ -0,0 +1,113 @@
...
+  const sendMessageToAuthor = async () => {
+    startSending();
+    try {
+      await sendMail(formData);
+    } catch (error) {
+      stopSending();
+    }
+    stopSending();
+    closeModal();
+  };
```
Comment (maximsan) [inline src/components/InTouchForm/InTouchForm.js:60]: it can be wrapped with useMemo

File: src/services/Azure.js [hunk 8]
```diff
@@ -283,4 +283,12 @@ export default class Azure {
     success: 2,
     fail: 3,
   };
+
+  sendMail = async (mailData) => {
+    try {
+      await axios.post(`https://dims-5.herokuapp.com/api/intouch`, mailData);
+    } catch (error) {
+      throw new Error("Can't send message. Please, try later.");
```
Comment (maximsan) [inline src/services/Azure.js:291]: don't forget to handle it with toaster

File: src/UI/CurrentUser/CurrentUser.js (hunk 1 above)
Comment (MaxDaniLOVE) [reply src/UI/CurrentUser/CurrentUser.js:6]: done

File: src/components/InTouchForm/InTouchForm.js (hunk 3 above)
Comment (MaxDaniLOVE) [reply src/components/InTouchForm/InTouchForm.js:56]: error handling implemented at redux actions

File: src/components/InTouchForm/InTouchForm.js (hunk 4 above)
Comment (MaxDaniLOVE) [reply src/components/InTouchForm/InTouchForm.js:73]: done

File: src/services/Azure.js (hunk 8 above)
Comment (MaxDaniLOVE) [reply src/services/Azure.js:291]: done

File: src/components/InTouchForm/InTouchForm.js (hunk 6 above)
Comment (MaxDaniLOVE) [reply src/components/InTouchForm/InTouchForm.js:33]: done

File: src/components/InTouchForm/InTouchForm.js (hunk 7 above)
Comment (MaxDaniLOVE) [reply src/components/InTouchForm/InTouchForm.js:60]: done

File: src/UI/CustomLinks/SocialNetworksLink.js (hunk 2 above)
Comment (MaxDaniLOVE) [reply src/UI/CustomLinks/SocialNetworksLink.js:7]: done

File: src/components/InTouchForm/InTouchForm.js (hunk 3 above)
Comment (maximsan) [reply src/components/InTouchForm/InTouchForm.js:56]: fine! :)

**Pull request merged**